tcms/testplans/models.py

```diff
@@ -151,7 +151,8 @@ def clone(  # pylint: disable=too-many-arguments
                 tc_src.clone(new_author, [tp_dest])
             else:
                 # otherwise just link the existing TC to the new TP
-                tp_dest.add_case(tc_src)
+                tc_plan = TestCasePlan.objects.filter(case=tc_src, plan=self).first()
+                tp_dest.add_case(tc_src, sortkey=tc_plan.sortkey)
```

Comment:
This is terribly inefficient b/c it makes 1 query for each test case we cycle through.  

Change the loop query to `self.cases.all().annotate(sortkey = F('testcaseplan__sortkey'))` and you can then access the value as `tc_src.sortkey`.

Reply:
@atodorov thanks you for the fast review ! Indeed my first patch was clearly inefficient, django queryset are really powerful. Unit tests has also shown that the clone with TC copy was also buggy.